### ISSUE TITLE
Simplify how we define QRHO, URHO, ...

### ISSUE THREAD

**Issue by zingale** [AMReX-Astro/Castro#267]
We have variables `QTHERM` and `NTHERM` that are used in defining `NVAR` and `QVAR`, but these do not really seem to be used or needed.  Keeping them around is confusing.

**Comment by maxpkatz** [comment]
QTHERM and NTHERM are useful for bookkeeping purposes inside ca_set_method_params, but I agree they do not need to be kept as module variables outside of the lifetime of that routine.

**Comment by zingale** [comment]
Currently we define `QRHO`, `QU`, ... and the conservative counterparts `URHO`, `UMX`, ... manually in `Castro_nd.F90` and set the total to `QVAR` and `NVAR`.  We also separately define `Density`, `Xmom`, ... in C++ in `Castro_setup.cpp` and rely on us doing these in the same order.  We should automate all of this.

E.g., define a text file `variables` of the form:

```
# C++ name     F90 name      C++ add-to          F90 add-to      preprocessor
Density             URHO            NUM_STATE       nvar
```

And a similar section for primitive variables, which I think are only needed in Fortran.

This file can be parsed by python to write include files for both C++ and Fortran.

**Comment by zingale** [comment]
We actually do the C++ first and pass the indices into Fortran to set the Fortran versions (offsetting by 1, because Fortran is 1-based).

**Comment by maxpkatz** [comment]
This is good by me. We actually want to do this for GPUs, because it helps a lot of NVAR, etc. are known at compile time.

**Comment by zingale** [comment]
I wasn't originally thinking of doing this at compile time, but rather statically, but we could do it at compile time.  The issue then, I think, is that it will basically always rebuild when we type `make` since we'll force the headers to be regenerated to capture any changes in the preprocessor.  

We might be able to cache the old files so they are not overwritten unless things change, in which case `make` would continue to do nothing after a successful build.

**Comment by maxpkatz** [comment]
I wasn't suggesting doing it at compile time versus statically. I was suggesting that if we do it statically, then NVAR will be known at compile time.

**Comment by zingale** [comment]
okay.  Yes, I think that can be done.

**Comment by zingale** [comment]
we can get rid of `ca_init_godunov_indices()` as part of this, since we will define everything together in `ca_set_method_params`

**Comment by zingale** [comment]
to make everything known at compile time, we need to fix #274 first
